transitions/transition_extraction.py

```
@@ -0,0 +1,211 @@
+
+    return p_tail, m_mid, n_head
+
+# Helper fn to deal with small or diff size audios
+def safe_crossfade_duration(a_path: Path, b_path: Path, desired_cross: float) -> float:
+    
+    # Returns a safe crossfade duration that won't exceed the length of either audio file.
+    
+    len_a = duration(str(a_path))
+    len_b = duration(str(b_path))
+
+    # pick the shortest file’s length * 0.8 to stay safe
```

Ruff (0.13.1)

Comment contains ambiguous `’` (RIGHT SINGLE QUOTATION MARK). Did you mean ``` (GRAVE ACCENT)?

(RUF003)